.github/workflows/main.yml

```diff
@@ -540,14 +540,33 @@ jobs:
           event-type: depthai-python-release
           client-payload: '{"ref": "${{ github.ref }}", "sha": "${{ github.sha }}"}'
 
+  # notify_hil_workflow_linux_x86_64:
+  #   needs: [build-linux-x86_64]
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - name: Repository Dispatch
+  #       uses: peter-evans/repository-dispatch@v2
+  #       with:
+  #         token: ${{ secrets.HIL_CORE_DISPATCH_TOKEN }}
+  #         repository: luxonis/depthai-core-hil-tests
+  #         event-type: python-hil-event
+  #         client-payload: '{"ref": "${{ github.ref }}", "sha": "${{ github.sha }}"}'
+
   notify_hil_workflow_linux_x86_64:
     needs: [build-linux-x86_64]
     runs-on: ubuntu-latest
     steps:
-      - name: Repository Dispatch
-        uses: peter-evans/repository-dispatch@v2
-        with:
-          token: ${{ secrets.HIL_CORE_DISPATCH_TOKEN }}
-          repository: luxonis/depthai-core-hil-tests
-          event-type: python-hil-event
-          client-payload: '{"ref": "${{ github.ref }}", "sha": "${{ github.sha }}"}'
\ No newline at end of file
+      - name: Dispatch an action and get the run ID
+        uses: codex-/return-dispatch@v1
+        id: return_dispatch
+        with:
+          token: ${{ secrets.HIL_CORE_DISPATCH_TOKEN }} # Note this is NOT GITHUB_TOKEN but a PAT 
+          ref: main # or refs/heads/target_branch
+          repo: depthai-core-hil-tests
+          owner: luxonis
+          workflow: regression_test.yml
+          workflow_inputs: '{"commit": "${{ github.ref }}", "sha": "${{ github.sha }}", "parent_url": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"}' 
+          workflow_timeout_seconds: 120 # Default: 300
```

Comment:
This is non-blocking dispatch right? Or does it wait for regression_test to finish?
Also, can we augment that to not be workflow specific instead?